pkg/plugins/runtime/k8s/controllers/pod_converter.go

```diff
@@ -50,24 +51,36 @@ func (p *PodConverter) PodToDataplane(
 }
 
 func (p *PodConverter) PodToIngress(ctx context.Context, zoneIngress *mesh_k8s.ZoneIngress, pod *kube_core.Pod, services []*kube_core.Service) error {
-	zoneIngressProto := &mesh_proto.ZoneIngress{}
-	// Pass the current dataplane so we won't override available services in Ingress section
-	if err := p.IngressFor(ctx, zoneIngressProto, pod, services); err != nil {
+	logger := converterLog.WithValues("ZoneIngress.name", zoneIngress.Name, "Pod.name", pod.Name)
+	// Start with the existing ZoneIngress spec so we won't override available services in Ingress section
+	zoneIngressRes := core_mesh.NewZoneIngressResource()
+	if err := p.ResourceConverter.ToCoreResource(zoneIngress, zoneIngressRes); err != nil {
+		logger.Error(err, "unable to convert ZoneIngress k8s object into core resource")
```

Comment:
I'd rather wrap an error that you return and log it at the top of call stack if needed instead of doing log + error.